ui/views_test.py

```diff
@@ -370,10 +391,118 @@ def test_collection_viewset_detail_as_superuser(logged_in_apiclient):
+    collection.save()
+    mock_moira_client.return_value.user_lists.return_value = [moira_list.name]
+    url = reverse('collection-detail', kwargs={'collection_key': collection.hexkey})
+    result = client.get(url)
+    assert result.status_code == status.HTTP_200_OK
+
+
+def test_collection_detail_view_permission(mock_moira_client, logged_in_apiclient):
+    """
+    Tests that a user cannot view a collection if user is a member of collection's view_lists
+    """
+    client, _ = logged_in_apiclient
+    collection = CollectionFactory()
+    moira_list = factories.MoiraListFactory()
+    collection.view_lists = [moira_list]
+    collection.save()
+    mock_moira_client.return_value.user_lists.return_value = [moira_list.name]
+    url = reverse('collection-detail', kwargs={'collection_key': collection.hexkey})
+    result = client.get(url)
+    assert result.status_code == status.HTTP_403_FORBIDDEN
+
+
+def test_video_detail_view_permission(mock_moira_client, logged_in_apiclient):
+    """
+    Tests that a user can view a video if user is a member of collection's view_lists
+    """
+    client, _ = logged_in_apiclient
+    video = VideoFactory()
+    collection = video.collection
+    moira_list = factories.MoiraListFactory()
+    collection.view_lists = [moira_list]
+    collection.save()
+    mock_moira_client.return_value.user_lists.return_value = [moira_list.name]
+    url = reverse('video-detail', kwargs={'video_key': video.hexkey})
+    result = client.get(url)
+    assert result.status_code == status.HTTP_200_OK
+    assert json.loads(result.context_data['js_settings_json'])['editable'] is False
+
+
+def test_video_detail_admin_permission(mock_moira_client, logged_in_apiclient):
+    """
+    Tests that a user can view a video if user is a member of collection's admin_lists
+    """
+    client, _ = logged_in_apiclient
+    video = VideoFactory()
+    collection = video.collection
+    moira_list = factories.MoiraListFactory()
+    collection.admin_lists = [moira_list]
+    collection.save()
+    mock_moira_client.return_value.user_lists.return_value = [moira_list.name]
+    url = reverse('video-detail', kwargs={'video_key': video.hexkey})
+    result = client.get(url)
+    assert result.status_code == status.HTTP_200_OK
+    assert json.loads(result.context_data['js_settings_json'])['editable'] is True
+
+
+def test_video_detail_no_permission(mock_moira_client, logged_in_apiclient):
+    """
+    Tests that a user cannot view a video if user is not a member of collection's lists
+    """
+    client, _ = logged_in_apiclient
+    video = VideoFactory()
+    collection = video.collection
+    moira_list = factories.MoiraListFactory()
+    collection.admin_lists = [moira_list]
+    collection.save()
+    mock_moira_client.return_value.user_lists.return_value = ['other_list']
+    url = reverse('video-detail', kwargs={'video_key': video.hexkey})
+    result = client.get(url)
+    assert result.status_code == status.HTTP_403_FORBIDDEN
```

Comment:
_NOTE: Sorry, I wrote a comment during the last round of reviews that didn't save for some reason. Repeating it here_

I know it's sort of the pattern in the file, but in these new test cases there's a lot of repetition of the sort that pytest is built to address. One thing we can/should do is fixture-ize commonly-needed scenarios. For example, a lot of these cases need a collection that a user will have admin or view permissions for. Instead of repeating that a bunch of times and obscuring the intent of the test, we can turn this idea into fixtures doing something like this:

```python
from types import SimpleNamespace

@pytest.fixture
def collection_data():
    video = VideoFactory()
    return SimpleNamespace(
        collection=video.collection,
        video=video
    )

@pytest.fixture
def admin_col_data(collection_data, mock_moira_client):
    moira_list = factories.MoiraListFactory()
    collection.admin_lists = [moira_list]
    collection.save()
    mock_moira_client.return_value.user_lists.return_value = [moira_list.name]
    return collection_data

@pytest.fixture
def viewable_col_data(collection_data, mock_moira_client):
    moira_list = factories.MoiraListFactory()
    collection.view_lists = [moira_list]
    collection.save()
    mock_moira_client.return_value.user_lists.return_value = [moira_list.name]
    return collection_data
    
def test_collection_detail_view_permission(logged_in_apiclient, viewable_col_data):
    client, _ = logged_in_apiclient
    url = reverse('video-detail', kwargs={'video_key': viewable_col_data.video.hexkey})
    result = client.get(url)
    assert result.status_code == status.HTTP_200_OK
    assert json.loads(result.context_data['js_settings_json'])['editable'] is False
    
def test_collection_detail_admin_permission(logged_in_apiclient, admin_col_data):
    client, _ = logged_in_apiclient
    url = reverse('video-detail', kwargs={'video_key': admin_col_data.video.hexkey})
    result = client.get(url)
    assert result.status_code == status.HTTP_200_OK
    assert json.loads(result.context_data['js_settings_json'])['editable'] is True

# etc, etc...
```
In that example, any test case that should involve a scenario where the user has admin permissions for the given collection would just need to add `admin_col_data` to the list of arguments.

[This test file in micromasters](https://github.com/mitodl/micromasters/blob/master/financialaid/views_pytest_test.py) has some of these patterns if you're interested. Using `SimpleNamespace` instead of multiple return values is particularly helpful and readable.